### ISSUE TITLE
Docs Issue - suggestion to clarify

### ISSUE THREAD

**Issue by kdog1425** [PrefectHQ/prefect#5523]
Hello!

I've found an issue here:`parallel pipelines.`

I believe this is a bit misleading - because the work is not done in parallel, unless the relevant executor is chosen. I suggest clarifying and adding a note which emphasises that to actually get parallel execution, one needs to set LocalDaskExecutor or DaskExecutor as the executor class. 

Page reference: https://docs.prefect.io/core/concepts/mapping.html

**Comment by Fraznist** [comment]
Instead of `creating independent parallel pipelines.`
Something like `creating identical, but independent, task pipelines`

I feel like introducing how to achieve actual parallelism via executors should be out of scope at that stage of the docs. Docs shouldn't make you jump around pages, especially forward jumps tend to be confusing for beginners. Removing the `parallel` keyword should eliminate the need to mention `Dask` or `executors`.

**Comment by kdog1425** [comment]
Makes sense to me @Fraznist - shall I open a PR?

**Comment by tpdorsey** [comment]
Resolved by https://github.com/PrefectHQ/prefect/pull/5587
